src/main/java/com/google/api/codegen/viewmodel/DescriptorConfigView.java

```diff
@@ -0,0 +1,82 @@
+/* Copyright 2017 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.viewmodel;
+
+import com.google.api.codegen.SnippetSetRunner;
+import com.google.auto.value.AutoValue;
+import java.util.List;
+import javax.annotation.Nullable;
+
+@AutoValue
+public abstract class DescriptorConfigView implements ViewModel {
+  public abstract String templateFileName();
+
+  public abstract List<PageStreamingDescriptorView> pageStreamingDescriptors();
+
+  @Nullable
+  public abstract List<BatchingDescriptorView> batchingDescriptors();
+
+  public abstract List<LongRunningOperationDetailView> longRunningDescriptors();
+
+  public abstract List<GrpcStreamingDetailView> grpcStreamingDescriptors();
+
+  public abstract String interfaceKey();
+
+  public abstract String outputPath();
+
+  public abstract boolean hasPageStreamingMethods();
+
+  public abstract boolean hasBatchingMethods();
+
+  public abstract boolean hasLongRunningOperations();
+
+  public boolean hasGrpcStreamingMethods() {
+    return grpcStreamingDescriptors().size() > 0;
```

Comment:
Can we be consistent with the `hasX` methods? Either have all of them be set manually (like hasPageStreamingMethods, hasBatchingMethods, hasLongRunningOperations) or have all of them check the length of the corresponding list field (like hasGrpcStreamingMethods)